ddtrace/debugging/_probe/remoteconfig.py

```diff
@@ -251,19 +251,42 @@ class ProbePollerEvent(object):
 ProbePollerEventType = int
 
 
-class ProbeRCAdapter(RemoteConfigCallBack):
+class DebuggerRemoteConfigSubscriber(RemoteConfigSubscriber):
     """Probe configuration adapter for the RCM client.
 
     This adapter turns configuration events from the RCM client into probe
     events that can be handled easily by the debugger.
     """
 
-    def __init__(self, callback):
-        # type: (Callable[[ProbePollerEventType, Iterable[Probe]], None]) -> None
-        self._callback = callback
+    def __init__(self, data_connector, callback, name):
+        super(DebuggerRemoteConfigSubscriber, self).__init__(data_connector, callback, name)
         self._configs = {}  # type: Dict[str, Dict[str, Probe]]
+        self._status_timestamp = time.time()
         self._next_status_update_timestamp()
 
+    def _exec_callback(self, data, test_tracer=None):
+        if data:
+            metadatas = data["metadata"]
+            rc_configs = data["config"]
+            # DEV: We emit a status update event here to avoid having to spawn a
+            # separate thread for this.
+            log.debug("[%s][P: %s] Dynamic Instrumentation Updated", os.getpid(), os.getppid())
+            for idx in range(len(rc_configs)):
+                if time.time() > self._status_timestamp:
```

Comment:
When wouldn't this be true? Can you explain it a bit?

Reply:
AFAIK, this condition depends of `config.diagnostics_interval` (1h).

Each time this condition is True, we update `self._status_timestamp` in this function:

https://github.com/DataDog/dd-trace-py/pull/5464/files/970621058ea1214d48946ae19fbe6746d1eb3dbd#diff-ae57ceee6484ce78fa6d58aef59ede86def77b5148eeb3a5a6dde71a56eba9fdR290
```
    def _next_status_update_timestamp(self):
        # type: () -> None
        self._status_timestamp = time.time() + config.diagnostics_interval
```

@P403n1x87, could you confirm this please?

Reply:
That's right. We want to re-emit probe status messages periodically, so at every iteration we set the new "deadline" in the future. When we are past that we re-emit the status messages.